tools/topology/sof-apl-wm8804.m4

```diff
@@ -52,8 +59,15 @@ DAI_ADD(sof/pipe-dai-playback.m4,
 	PIPELINE_SOURCE_1, 2, s24le,
 	1000, 0, 0, SCHEDULE_TIME_DOMAIN_DMA)
 
+# capture DAI is SSP5 using 2 periods
+# Buffers use s24le format, 1000us deadline on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-capture.m4,
+	2, SSP, 5, SSP5-Codec,
+	PIPELINE_SINK_2, 2, s24le,
+	1000, 0, 0, SCHEDULE_TIME_DOMAIN_DMA)
```

Comment:
@plbossart the changes look good. Just out of curiosity, why is this SCHEDULE_TIME_DOMAIN_DMA and not SCHEDULE_TIME_DOMAIN_TIMER. Whats the rule of thumb for using these. I'm wondering if this is documented somewhere.

Reply:
I guess we need to document this in sof-docs.

Reply:
When the codec is clock slave, the firmware does not need to track the DMA interrupts - they are by definition synchronous with the system timer, so all pipelines can be scheduled using a timer.

When the codec is master, the DMA interrupts are synchronous with the codec clock, not the internal SOC timers, so we cannot use the timer scheduling for such pipelines.

I think the rule is that scheduling should be DMA based by default, unless all clocks are synchronous with the internal timers, in which case we can skip the DMA interrupt handling.

As discussed in other thread, the 'agent' should also be improved to know that at least one pipeline is not using the timer scheduling.